sql/parser2/eval.go

```diff
@@ -661,7 +662,30 @@ func evalUnaryExpr(expr *UnaryExpr, env Env) (Datum, error) {
 }
 
 func evalFuncExpr(expr *FuncExpr, env Env) (Datum, error) {
-	return null, fmt.Errorf("TODO(pmattis): unsupported expression type: %T", expr)
+	name := strings.ToLower(expr.Name.String())
+	b, ok := builtins[name]
+	if !ok {
+		return null, fmt.Errorf("%s: unknown function", expr.Name)
+	}
+	if b.nArgs != -1 && b.nArgs != len(expr.Exprs) {
```

Comment:
Yeah, it seemed better to centralize this check, though I imagine this will evolve as we add more functions.